woocommerce/woocommerce-ios
# [Bundle products in order] Display bundle child item prices according to whether they are priced individually

Closes: #11250
⚠️ Depends on #11316 ⚠️ 


## Why
If child items aren't priced individually, we want to set the price to $0.00 with a grey text colour.

Note: We discussed in pe5pgL-42P-p2#comment-3289 using the regular price for the unit price, and displaying it with a strikethrough. However, when I tried this (see ba9f724f663cba46bcd6d2933ee59cbb66b7a15c) it both added a lot of complexity using attributed strings and changed the line height for the price line because of how we create the `AttributedText` view in SwiftUI. I'd suggest we stick with the $0.00 price for the unit price for now, but I'm interested in your perspective on whether it's worth doing the strikethrough.

## How
Using the `pricedIndividually` property added in #11316:

* We add `pricedIndividually` to `ProductRowViewModel`. If a product or variation is not priced individually, the `price` is set to `"0"`.
* In `EditableOrderViewModel` we set `pricedIndividually` when the `ProductRowViewModels` are created, based on whether the bundle items that correspond to each child item in the order are priced individually.
* In `CollapsibleProductCardPriceSummary` we use `ProductRowViewModel.pricedIndividually` to make all of the price text grey if the row is not priced individually.

## Testing instructions
Prerequisite: the store has [Product Bundles extension](https://woocommerce.com/products/product-bundles/) enabled, and also a bundle product with at least one bundled item priced individually and one bundled item _not_ priced individually.

- Go to the Orders tab
- Tap `+` to create an order
- Tap `Add Products` and tap a non-bundle product
- Tap a bundle product from the prerequisite
- Configure the bundle
- Add the products to the order
- On the order form, confirm the parent bundle and any child items that are priced individually show the regular price as before, while any child items _not_ priced individually show a $0.00 price in grey.

## Screenshots
Before|After
-|-
![Before-Collapsed](https://github.com/woocommerce/woocommerce-ios/assets/8658164/1f413bad-df0e-4253-869b-9ba69df329a7)|![After-Collapsed](https://github.com/woocommerce/woocommerce-ios/assets/8658164/00dc248f-7194-4dd1-a581-50f6db6a76e8)
![Before-Expanded](https://github.com/woocommerce/woocommerce-ios/assets/8658164/836e316a-3793-4937-b497-824e6b08bf6a)|![After-Expanded](https://github.com/woocommerce/woocommerce-ios/assets/8658164/207014ab-91e3-4fab-bf33-339407ad6796)


---
